src/overlay025/poketch_system.c

```diff
@@ -0,0 +1,790 @@
+
+    new_system = Heap_AllocFromHeap(7, sizeof(PoketchSystem));
+
+    if (new_system != NULL) {
+        *poketchSys = new_system;
+
+        new_system->fieldSys = fieldSys;
+        new_system->saveData = saveData;
+        new_system->poketchData = SaveData_PoketchData(saveData);
+        new_system->bgl = param3;
+        new_system->unk_50 = param4;
+
+        if (PoketchSystem_InitInternal(new_system)) {
+            sub_0201E3D8();
+            sub_0201E450(4);
+
+            new_system->poketchSysPtr = poketchSys;
+            new_system->unk_38 = SysTask_Start(ov25_02253E94, new_system, 4);
+
+            SysTask_Start(PoketchSystem_MainTask, new_system, 0);
+        }
+    }
+}
+
+void ov25_02253D5C (PoketchSystem *poketchSys)
+{
+    if ((poketchSys->unk_00 != 3) && (poketchSys->unk_00 != 4)) {
+        ov25_02253E9C(poketchSys, 3);
+    }
+}
+
+BOOL ov25_02253D70 (PoketchSystem *poketchSys)
+{
+    return poketchSys == NULL;
+}
+
+void ov25_02253D7C (PoketchSystem *poketchSys, int param1, u32 param2)
+{
+    switch (param1) {
+    case 0:
+        break;
+    case 1:
+        poketchSys->unk_04 = 1;
+        break;
+    case 5:
+    {
+        u32 step_count = PoketchData_StepCount(poketchSys->poketchData);
+
+        if (++step_count > 99999) {
+            step_count = 0;
+        }
+
+        PoketchData_SetStepCount(poketchSys->poketchData, step_count);
+        poketchSys->pedometerUpdated = TRUE;
+    }
+    break;
+    case 4:
+        if (poketchSys->unk_02 == 2) {
+            if (poketchSys->unk_44) {
+                poketchSys->unk_44(poketchSys->unk_48);
+            }
+        }
+        break;
+    }
+}
+
+BOOL ov25_02253DD4 (PoketchSystem *poketchSys)
+{
+    return poketchSys->unk_03;
+}
+
+enum PoketchAppID PoketchSystem_CurrentAppID(PoketchSystem *poketchSys)
+{
+    return poketchSys->loadedAppID;
+}
+
+static BOOL PoketchSystem_InitInternal (PoketchSystem *poketchSys)
+{
+    if (ov25_02254560(&(poketchSys->unk_1C), &(poketchSys->unk_20), poketchSys->unk_50, poketchSys)) {
+        poketchSys->unk_00 = 0;
+        poketchSys->unk_01 = 0;
+        poketchSys->unk_03 = 0;
+        poketchSys->unk_04 = 0;
+        poketchSys->pedometerUpdated = FALSE;
+        poketchSys->unk_05 = 0;
+        poketchSys->unk_06 = 0;
+        poketchSys->loadedAppID = -1;
+        poketchSys->unk_02 = 0;
+        poketchSys->unk_08 = 0;
+        poketchSys->buttonDir = BUTTON_UP;
+
+        if (ov25_02254284(poketchSys)) {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+static void ov25_02253E20 (PoketchSystem * poketchSys)
+{
+    SysTask_Done(poketchSys->unk_38);
+
+    ov25_022542C8(poketchSys);
+    PoketchSystem_UnloadApp(poketchSys);
+    ov25_02254754(poketchSys->unk_1C);
+}
+
+static void PoketchSystem_MainTask(SysTask *task, void *system)
+{
+    static void(*const v0[])(PoketchSystem *) = {
+        ov25_02253EA4,
+        ov25_02253F2C,
+        ov25_0225406C,
+        ov25_022540D8
+    };
```

Comment:
Odd construction. Can we declare a one-time-use function-pointer typedef to make it clearer what this table is meant to represent?